src/niMidi.cpp

```diff
@@ -234,21 +311,58 @@ class NiMidiSurface: public BaseSurface {
 			this->_sendSysex(CMD_TRACK_NAME, 0, numInBank, name);
 			// todo: level meters, volume, pan
 		}
-		// todo: navigate tracks, navigate banks
+		// todo: navigate tracks, navigate banks. NOTE: probably not here
+	}
+
+	void ClearSelected() {
+		// Clear all selected tracks. Copyright (c) 2010 and later Tim Payne (SWS)
+		int iSel = 0;
+		for (int i = 0; i <= GetNumTracks(); i++) // really ALL tracks, hence no use of CSurf_NumTracks
```

Comment:
I've never quite understood this difference between CSurf_NumTracks and GetNumTracks. I assume CSurf_NumTracks excludes tracks that aren't shown in the TCP/MCP depending on the argument?

Reply:
My understanding is the following: GetNumTracks() always yields the same result as CSurf_NumTracks(false), i.e. TCP view. TCP view always counts all tracks. In MCP view, however, you can hide tracks which is very useful in a professional mixing environment. Thus, it is desirable for control surfaces which include a mixer to reflect that very view and thus track numbering logic. I don't know if we will ever use it as it may create inconsistencies with the track/instance focus functionality of KK (which is one of the main benefits of that keyboard/plugin combo). I introduced GetNumTracks for this particular function ClearSelected() rather than using Csurf_NumTracks(false) because here we really do not have an option to choose MCP numbering logic. The track focus of KK only works meaningfully if only one track is selected. I had experimented with allowing multiple tracks to be selected when I implemented the top row buttons, but the experience in combination with track focus is nothing but confusing. Also the screens do not reflect multiple selects properly. NI seems to assume that user will always only select one track (which is a bit stupid but well that's how they implemented it).